### ISSUE TITLE
RNN Controller is implemented differently from the original paper

### ISSUE THREAD

**Issue by Kipsora** [titu1994/neural-architecture-search#8]
In the paper, the RNN is described as: "Every prediction is carried out by a softmax classifier and then fed into the next time step as input". However, I found in the implementation, the input is the output of **previous epoch** rather than the output of **previous time step**. Actually, I am confused with the initial state of RNN. The paper didn't state what the initial state is.

Related codes:
https://github.com/titu1994/neural-architecture-search/blob/d5f5c9d624026263796984c9b71e4ec6443b61b4/train.py#L67-L95

**Comment by titu1994** [comment]
The graph of the rnn is built such that in one pass, all the different classifiers will output sequentially based on each other's last state. 

For the first state, it is common to use a zero state vector.
